expression/expression.go

```diff
@@ -687,8 +688,106 @@ func IsBinaryLiteral(expr Expression) bool {
 	return ok && con.Value.Kind() == types.KindBinaryLiteral
 }
 
-// CheckExprPushFlash checks a expr list whether each expr can be pushed to flash storage.
-func CheckExprPushFlash(exprs []Expression) (exprPush, remain []Expression) {
+// CheckExprPushDown check if all the expr in the expr list can be push down to the storage.
+func CheckExprPushDown(sc *stmtctx.StatementContext, exprs []Expression, client kv.Client, storeType kv.StoreType) bool {
+	_, _, remained := ExpressionsToPB(sc, exprs, client)
+	if len(remained) > 0 {
+		return false
+	}
+	switch storeType {
+	case kv.TiFlash:
+		return checkFlashExprPushDown(exprs)
+	default:
+		return checkTiKVExprPushDown(exprs)
+	}
+}
+
+// ExprPushDown push down an expr list to storage.
+// Note this function assumes all the expr in expr list can be convert to pb expr
+func ExprPushDown(sc *stmtctx.StatementContext, exprs []Expression, client kv.Client, storeType kv.StoreType, skipPBCheck bool) (exprPush, remain []Expression) {
+	if !skipPBCheck {
+		_, exprPush, remain = ExpressionsToPB(sc, exprs, client)
+		var storeRemained []Expression
+		switch storeType {
+		case kv.TiFlash:
+			exprPush, storeRemained = flashExprPushDown(exprPush)
+		default:
```

Comment:
We can remove these lines and reuse line 721~725?

Reply:
addressed